### ISSUE TITLE
Upgrade Symfony to a version that is maintained

### ISSUE THREAD

**Issue by phavekes** [OpenConext/OpenConext-engineblock#1516]
[This issue is imported from pivotal - Originaly created at Aug 15, 2018 by Michiel Kodde](https://www.pivotaltracker.com/story/show/159790180)

Symfony will stop supporting 2.8  Nov 2018 for bug fixes and security fixes will stop the year after that.

Upgrading to Symfony 3.4 is our best option right now as we are not yet ready to drop PHP 5 support.

**Comment by phavekes** [comment]
[PR can be found here](https://github.com/OpenConext/OpenConext-engineblock/pull/590), it has surpassed it\'s POC stage at this point as it looks like the upgrade is very feasable. The last obstacles for a passing travis build are:

```
     [exec]     /home/travis/build/OpenConext/OpenConext-engineblock/src/OpenConext/EngineBlockFunctionalTestingBundle/Features/AttributeAggregation.feature:39
     [exec]     /home/travis/build/OpenConext/OpenConext-engineblock/src/OpenConext/EngineBlockFunctionalTestingBundle/Features/ClearErrorMessages.feature:93
     [exec]     /home/travis/build/OpenConext/OpenConext-engineblock/src/OpenConext/EngineBlockFunctionalTestingBundle/Features/Encryption.feature:26
``` (Michiel Kodde - Aug 21, 2018)

**Comment by phavekes** [comment]
@thijskh, the Symfony upgrade is passing builds on Travis! It might be a good idea to ship this in the next EB release (5.9) giving us the oportunity to further test and verify the upgrade. What are your thougths on this? (Michiel Kodde - Aug 23, 2018)

**Comment by phavekes** [comment]
# Roll back test log
This is the summary of what I\'ve performed as a functional test of this PR.

The first step will be to create an environment that represents EB in production. The downgrade path will now be more painfull as there is no rollback feature in place yet.

## Downgrade working version to latest 5.7
To simualate the way the update will be executed, first perform a donwgrade to 5.7.6

PWD is assumed to be: `/opt/openconext/Openconext-engineblock`

1. Backup current working database `$ sudo mysqldump -h localhost eb > ./eb-dump-5.8.sql`
1. Checkout master: `$ git checkout master`
1. $ composer install` :coffee:
1. Verify engine is up and running
1. Downgrade to the correct working state
   1. `$ app/console doctrine:migrations:execute 20180804090135 --down`
   1. `$ app/console doctrine:migrations:execute 20180724175453 --down`
   1. `$ app/console doctrine:migrations:execute 20180618134736 --down`
   1. `$ app/console doctrine:migrations:status` Should show:
    ```
    >> Executed Migrations:                                14
    >> Executed Unavailable Migrations:                    0
    >> Available Migrations:                               17
    >> New Migrations:                                     3
    ```
1. Downgrade to 5.7.1 `$ git checkout 5.7.6`
1. Set the required settings in `application.ini` or `engineblock.ini`
    ```
    engineApi.users.janus.username = "some user"
    engineApi.users.janus.password = "some password"
    engineApi.users.profile.username = "some user"
    engineApi.users.profile.password = "some password"
    ```
1. $ composer install` :coffee:
1. Restart PHP FPM ensuring any cahced PHP files are cleaned up. `$ sudo service php-fpm restart`
1. Verify engine is up (visit engine.vm.openconext.org and you should see the homepage)
1. It might be required to further tweak your config settings, or reload some services (shibd, httpd, php-fpm)
1. Empty `sps_entity_ids_without_consent` 
   1. `$ sudo mysql eb`
   1. `MariaDB [eb]> UPDATE sso_provider_roles_eb5 SET sps_entity_ids_without_consent = NULL;`
1. Push metadata to from Manage to EB. Take your favourite route to push new metadata
   1. Use the manage backdoor option
   1. Push metadata directly onto Engine API (curl/postman) (you can use the metadata dump below)
   1. Use Ansible playbook `$ ansible-playbook -i environments/vm/inventory -u vagrant -e secrets_file=environments/vm/secrets/vm.yml --tags vm_only_provision_manage_eb provision-vm.yml` (on your host)
1. Verify engine is up and running, you should now be able to hande authentications.
1. A 5.7 dump might prove usefull later: `$ sudo mysqldump -h localhost eb > ./eb-dump-5.7.sql`

You should now be at 5.7.6!

## Rolling upgrade 5.7.6 -> 5.8.3

Now we can start testing the rolling upgrade feature.

1. Checkout master: `$ git checkout master`
1. Cherry pick fbec1d3 (Add rolling update 5.8 compatibility support) `$ git cherry-pick fbec1d3` :warning: verify comit hash, might have been amended/smashed/fixed-up.
1. Set the required settings in `application.ini` and or `engineblock.ini` (for example set the `engineApi.users.metadataPush.username` and `engineApi.users.metadataPush.password`)
1. `$ composer install`
1. Run migrations
   1. `$ app/console doctrine:migrations:status` Should show:
    ```
    >> Executed Migrations:                                14
    >> Executed Unavailable Migrations:                    0
    >> Available Migrations:                               16
    >> New Migrations:                                     2
    ```
   1. `$ sudo mysqldump -h localhost eb > ./eb-dump-5.8.3-pre-migration.sql`
   1. `$ app/console doctrine:migrations:migrate`
1. Push metadata (I needed to restart Java apps before being able to enter Manage)
   1. You should be able to push from Manage
   1. Optionally push metadata with CURL/Postaman.
1. Verify engine is up and running.

## Rollback 5.8.3 -> 5.7.6

And the rollback should be very easy now compared to the previous steps required.

1. Verify EB is running and well
1. Downgrade to the correct working state
   1. `$ app/console doctrine:migrations:execute 20180804090135 --down`
   1. `$ app/console doctrine:migrations:execute 20180618134736 --down`
1. Downgrade to 5.7.1 `$ git checkout 5.7.6`
1. Set the required settings in `application.ini` or `engineblock.ini`
1. Composer install
1. Restart PHP FPM ensuring any cahced PHP files are cleaned up. `$ sudo service php-fpm restart`
1. Verify engine is up and running (without having pushed new metadata)

## Appendix: Postman export
See attachment, update credentials according to your eb.ini settings for the MetadataPush API settings. (Michiel Kodde - Sep 11, 2018)

**Comment by phavekes** [comment]
(Michiel Kodde - Sep 11, 2018)

**Comment by phavekes** [comment]
This chore took care of the not yet merged Symfony 3.4 upgrade #161282428 (Michiel Kodde - Feb 7, 2019)
